server/prisma/seed/seed.ts

```diff
@@ -0,0 +1,46 @@
+import { classes, classrooms, subjects, teachers, timetables, user } from "./data/index";
+import { PrismaClient } from "@prisma/client";
+
+const prisma = new PrismaClient();
+
+async function main() {
+  const user = await createUser();
+
+  await prisma.$transaction([
+    prisma.timetable.createMany({
+      data: timetables.map((timetable) => ({ ...timetable, userId: user.id })),
+    }),
+    prisma.classroom.createMany({
+      data: classrooms.map((classroom) => ({ ...classroom, userId: user.id })),
+    }),
+    prisma.teacher.createMany({
+      data: teachers.map((teacher) => ({ ...teacher, userId: user.id })),
+    }),
+    prisma.subject.createMany({
+      data: subjects.map((subject) => ({ ...subject, userId: user.id })),
+    }),
+    prisma.class.createMany({
+      data: classes.map((schoolClass) => ({ ...schoolClass, userId: user.id })),
+    }),
+  ]);
+}
+
+async function createUser() {
+  const userExists = await prisma.user.findFirst();
+
+  if (userExists) throw Error("User already exits. Please, change user seed data.");
+
+  return await prisma.user.create({
+    data: user,
+  });
+}
+
+main()
+  .catch((error) => {
+    console.log(error);
+    process.exit(1);
```

Comment:
If the code exits here, does it ever enter the `finally` callback?

Reply:
It depends. In case of error it won't because it will enter .catch but in case of success it will. I can rewrite it as .then().catch() and disconnect in .then(). That is the only way that I can think of now for better readability.

Reply:
It depends. In case of error it won't because it will enter .catch but in case of success it will. I can rewrite it as .then().catch() and disconnect in .then(). That is the only way that I can think of now for better readability.